src/read.jl

```diff
@@ -104,113 +104,160 @@ function read_storage(filepath)::Storage
+        )
+
+        # Try assigning the different demand flexibility parameters from the file
+        for k in keys(demand_flexibility_params_errs)
+            try
+                demand_flexibility[k] = demand_flexibility_parameters[1, k]
+            catch e
+                println(demand_flexibility_params_errs[k])
+            end
+        end
+
     catch e
-        println("Demand flexibility profiles not found in " * filepath)
-        demand_flexibility["flex_amt"] = nothing
-        demand_flexibility["enabled"] = false
-        demand_flexibility["duration"] = nothing
-        demand_flexibility["interval_balance"] = false
-        demand_flexibility["rolling_balance"] = false
-        demand_flexibility["cost_up"] = nothing
-        demand_flexibility["cost_dn"] = nothing
+        println("Demand flexibility parameters not found in " * filepath)
+        println(
+            "Demand flexibility parameters will default to allowing demand flexibility "
+            * "to occur."
+        )
+        demand_flexibility["enabled"] = "not_specified"
     end
 
-    # Set the demand flexibility costs and parameters
-    if demand_flexibility["enabled"]
-        # Pre-specify the demand flexibility parameters
-        demand_flexibility["duration"] = nothing
-        demand_flexibility["interval_balance"] = true
-        demand_flexibility["rolling_balance"] = true
-
-        # Try loading the demand flexibility parameters
-        demand_flexibility_parameters = DataFrames.DataFrame()
-        try
-            demand_flexibility_parameters = CSV.File(
-                joinpath(filepath, "demand_flexibility_parameters.csv")
-            ) |> DataFrames.DataFrame
-            println("...loading demand flexibility parameters")
-
-            # Create a dictionary to hold the error messages relevant to loading the 
-                # demand flexibility parameters
-            demand_flexibility_params_errs = Dict()
-            demand_flexibility_params_errs["duration"] = (
-                "The demand flexibility duration parameter is not defined. Will "
-                * "default to nothing."
-            )
-            demand_flexibility_params_errs["enabled"] = (
-                "The parameter that indicates if demand flexibility is enabled is not "
-                * "defined. Will default to being enabled."
-            )
-            demand_flexibility_params_errs["interval_balance"] = (
-                "The parameter that indicates if the interval load balance constraint "
-                * "is enabled is not defined. Will default to being enabled."
-            )
-            demand_flexibility_params_errs["rolling_balance"] = (
-                "The parameter that indicates if the rolling load balance constraint "
-                * "is enabled is not defined. Will default to being enabled."
-            )
+    # Check the feasibility of the duration parameter
+    if demand_flexibility["duration"] == nothing
```

Comment:
Would it be simpler is we default this parameter to `interval` at the beginning?

Reply:
Yeah, probably. That could remove the need for this `duration` check. I can make that change

Reply:
Done.